src/ngraph/node.cpp

```diff
@@ -104,48 +104,36 @@ std::shared_ptr<Node>
     Node::copy_with_new_inputs(const OutputVector& inputs,
                                const std::vector<std::shared_ptr<Node>>& control_dependencies) const
 {
-    shared_ptr<Node> clone;
-    if (is_type<op::GetOutputElement>(this))
+    shared_ptr<Node> clone = clone_with_new_inputs(inputs);
+    for (auto& cdep : control_dependencies)
     {
-        auto& value = inputs.at(0);
-        clone = make_shared<op::GetOutputElement>(value.get_node_shared_ptr(), value.get_index());
+        clone->add_control_dependency(cdep);
     }
-    else
+    return clone;
+}
+
+std::shared_ptr<Node> Node::copy_with_new_args(const NodeVector& args) const
+{
+    NODE_VALIDATION_CHECK(
```

Comment:
This check is a NOP since the string is the second arg thus it is the condition.